appinha/vizu
# feat: create custom user entity in django

## 💭 Motivation

Django's basic User entity works and is great, but it has a couple of flaws:

1. it uses `username` as main identifier instead of `email`. This used to make sense back in 2010, but not anymore;
2. if we want to add or remove attributes, it's not straightforward. The usual recommended way is to create a new `Profile` entity that points to User and add things there.

Adding to that, swapping the default user model is very difficult after you create the initial migrations, so it's a good idea to do this early on.


## 🗒️ Description

This PR creates a new `accounts` app and, within it, a new `CustomUser` entity to serve as the de-facto User model in our application.
The CustomUser model has the email as its main identifier, and as such is case-insensitive, unique and cannot be empty.


## 🛠️ Testing

1. open a terminal
2. cd into `backend`
3. run `poetry run python manage.py migrate`
4. run `poetry run python manage.py createsuperuser` and fill the required info
5. with the Django server up, visit `localhost:8000/admin` and log in with the superuser you just created. It should ask for your email and password
6. after logging in, click Users link, inside Accounts
7. you should see and be able to search for and edit any users you created.
